etna/ciim/client.py

```diff
@@ -1,85 +1,266 @@
+        web_reference: Optional[str] = None,
+        stream: Optional[Stream] = None,
+        sort: Optional[Sort] = None,
+        sort_order: Optional[SortOrder] = None,
+        template: Optional[Template] = None,
+        show_buckets: Optional[bool] = None,
+        buckets: Optional[list[str]] = None,
+        topics: Optional[list[str]] = None,
+        references: Optional[list[str]] = None,
+        offset: Optional[int] = None,
+        size: Optional[int] = None,
+    ) -> dict:
+        """Make request and return response for Kong's /fetch endpoint.
 
-        logger.debug(f"Response from Kong: {json}")
+        Search all metadata by keyword or web_reference. Results can be
+        bucketed, and the search restricted by bucket, reference, topic and
+        data stream. If both keyword and web reference are not provided, the
+        returned items will be empty.
 
-        if "message" in json:
-            raise KubernetesError(json["message"])
+        Keyword arguments:
 
-        if "error" in json:
-            raise KongError(f"Kong returned status {json['status']}")
+        keyword:
+            String to query all indexed fields
+        web_reference:
+            Return matches on references_number
+        stream:
+            Restrict results to given stream
+        sort:
+            Field to sort results.
+        sortOrder:
+            Order of sorted results
+        template:
+            @template data to include with response
+        show_buckets:
+            Include Etna search buckets with response
+        buckets:
+            Restrict results to given bucket(s)
+        topics:
+            Restrict results to given topic(s)
+        references:
+            Restrict results to given reference(s)
+        offset:
+            Offset for results. Mapped to 'from' before making request
+        size:
+            Number of results to return
+        """
+        params = {
+            "keyword": keyword,
+            "webReference": web_reference,
+            "stream": stream,
+            "sort": sort,
+            "sortOrder": sort_order,
+            "template": template,
+            "showBuckets": show_buckets,
+            "buckets": format_list_param(buckets),
+            "topics": format_list_param(topics),
+            "references": format_list_param(references),
+            "from": offset,
+            "size": size,
+        }
 
-        return json
-
-    def search(self, **kwargs):
-
-        if term := (
-            kwargs.pop("iaid", None)
-            or kwargs.pop("reference_number", None)
-            or kwargs.pop("term", None)
-            or ""
-        ):
-            kwargs["term"] = term
-
-        # In Python 'from' cannot be a kwargs. Map 'start' to 'from' before making request
-        kwargs["from"] = kwargs.pop("start", 0)
+        # remove empty values to make logged requests cleaner.
+        params = {k: v for k, v in params.items() if v is not None}
```

Comment:
This line of code "params = {k: v for k, v in params.items() if v is not None}"  is repeated in other places , can we not make it a function ?

Reply:
Maybe this code  if any({k: v for k, v in kwargs.items() if v is None}): in the SearchManager is no longer needed since the validation is in the KongClient.

Reply:
Cheers @mozaki1 

Good point, I think as soon as cleaning the params becomes more complex, we can move to a function. I think since the code is straightforward single line, it's ok as is for now.

Reply:
I would suggest having this sort of 'prep' done in another method... a little bit like what I've done here: https://github.com/nationalarchives/ds-judgments-public-ui/pull/16/files#diff-cdccc6f4bc34c68c2f90781a4cded97dcf7058208ff15098c6db2f9501fd9